### PR TITLE
breaking(api): section data rework

### PR DESCRIPTION
**Pull request by bobaaaaa**
These changes represents the current state of our section ui. Some notable changes:

- no recursive stages anymore. only two level stages.
- explicit companion items
- no explicit width for stage items. handled via stage_layout

The docs are wip.

### CODE REVIEW

File: stroeer/page/section/v1/section_page.proto [hunk 1]
```diff
@@ -61,49 +67,53 @@ message StageConfiguration {
...
+  // * `stage_layout`: represents a visual layout, which may affect teasers, colors, and other styles.
+  // If entry `stage_layout` is missing, use layout `stream`.
+  // the stage_layout field can be used to configure full page width commercials (e.g. nativendo, t-online empfiehlt)
   map<string, string> fields = 1;
+  repeated TeaserStageItem teaser_stage_items = 2;
+}
+
+// A single teaseable item.
+// alternatives: ClickableItem, LinkedItem, EverythingButAStage
+message TeaserStageItem {
```
Comment (bobaaaaa) [inline stroeer/page/section/v1/section_page.proto:92]: Some better naming ideas? Our first draft was `SubStage` + `SubStageItem`

Comment (zenderol) [inline stroeer/page/section/v1/section_page.proto:92]: I think the name "Teaser" implies that only an Article Teaser can be used here, I'd prefer "SubStage" naming or something like "ChildStage" or "StageChildItem"

Comment (harryurban) [reply stroeer/page/section/v1/section_page.proto:92]: I know what you mean, but doesn't a Commercial tease a product?

Comment (zenderol) [reply stroeer/page/section/v1/section_page.proto:92]: It does, but this is not our Business Domain. If a user clicks on a Teaser, he is not using our Page anymore. A "Teaser" only teases stuff from our domain, in my opinion.

Comment (harryurban) [reply stroeer/page/section/v1/section_page.proto:83]: // alternatives: ClickableItem, LinkedItem, EverythingButAStage

**Review by harryurban — APPROVED** [review]
❤️

**Comment by harryurban** [comment]
friendly reminder: https://github.com/stroeer/tapir/issues/229 (because of go failing)

File: stroeer/page/section/v1/section_page.proto [hunk 2]
```diff
@@ -59,51 +60,49 @@ message StageConfiguration {
   repeated stroeer.core.v1.Reference references = 2;
 }
 
-// A single generic item of a stage.
+// A generic model to represent what can be part of a stage.
 message StageItem {
```
Comment (thisismana) [inline stroeer/page/section/v1/section_page.proto:64]: this is virtually the same as `message TeaserStageItem` only difference is an allowed/forbiden field implemented as `oneof`. And again, `Commercial` and `ArticleTeaser` only differ in one field, but is a different message and is using a different mechanism (separate message).
This feels a bit inconsistent to me.

Comment (bobaaaaa) [reply stroeer/page/section/v1/section_page.proto:64]: Yep you're right. What would your proposal to remove the recursive stages after level 2?

File: stroeer/page/section/v1/section_page.proto [hunk 3]
```diff
@@ -38,18 +38,19 @@ message Section {
   stroeer.core.v1.Reference section_tree = 2;
 }
 
-// A block of teasers - can consist of editorial articles, advertisement and/or stages.
+// A "row" of teasers - can consist of editorial articles, advertisement and/or stages.
 message Stage {
-  repeated StageItem stage_items = 1;
-  StageConfiguration stage_configuration = 2;
+  StageConfiguration stage_configuration = 1;
```
[the remark this answers is not in the document]
Comment (bobaaaaa) [reply stroeer/page/section/v1/section_page.proto:43]: Unfortunately there're no namespaces in proto. On this file this looks verbose but on consumer usage view this is ok I guess

[the remark this answers is not in the document]
Comment (thisismana) [reply stroeer/page/section/v1/section_page.proto:43]: i mean the field name, not the message name. I've tested something with "namespaces", check #235 for better enums by just placing them into their corresponding message

File: stroeer/page/section/v1/section_page.proto (hunk 2 above)
Comment (thisismana) [reply stroeer/page/section/v1/section_page.proto:64]: I do not like this move, instead we can somehow guarantee this by contract and get rid of this duplication?

File: stroeer/page/section/v1/section_page.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (thisismana) [reply stroeer/page/section/v1/section_page.proto:43]: and then there is the possibility of nesting https://developers.google.com/protocol-buffers/docs/proto3#nested

we can move virtually everything into `message Stage` and see, how this works out

File: stroeer/page/section/v1/section_page.proto [hunk 4]
```diff
@@ -59,51 +60,49 @@ message StageConfiguration {
...
 
-// A single generic item of a stage.
+// A generic model to represent what can be part of a stage.
 message StageItem {
-  // Width instructions for rendering the single item.
-  StageItemWidth stage_item_width = 1;
-  // At most one field will be set at the same time.
-  // The delegate of the StageItem can be another Stage, an editorial article or a Commercial.
+  // Optional flags/hints for customization independent of the stage_item_type
+  map<string, string> fields = 1;
```
Comment (thisismana) [inline stroeer/page/section/v1/section_page.proto:66]: maybe reserve some space here because its expected that the `oneof` will grow in the future

**Pull request closed without merging**